src/java.base/unix/native/libjli/java_md.c

```diff
@@ -276,6 +276,16 @@ CreateExecutionEnvironment(int *pargc, char ***pargv,
                            char jrepath[], jint so_jrepath,
                            char jvmpath[], jint so_jvmpath,
                            char jvmcfg[],  jint so_jvmcfg) {
+    /* Compute/set the name of the executable. This is needed for macOS. */
```

Comment:
But this file is not used when on macOS, is it? (cfr java_md_macosx.m) )

Reply:
Hm. I think both are used? I'll need to double-check that.

Reply:
You are right. Don't know how this got in here, I must have thought it needed at some point when trying to get this to work on macOS. 

This really is a sign that I've been doing cargo-cult programming. The entire JDK bootstrapping mechanism is so spread-out and difficult to follow that it ends up being more "poke a bit here and see if it works" kind of programming instead of understanding what needs to be done and doing it. We should really make a follow-up refactoring on this to get it to a better state, across all modules and components...